### PR TITLE
iOS SDK Journey

### PR DESCRIPTION
**Pull request by KellieHawks**
This iOS integration guide takes you on a step-by-step journey on setup best practices when first integrating the iOS SDK and its core components in your application. This guide will help you decouple any dependencies on the Braze iOS SDK from the rest of your production code.

### CODE REVIEW

**Comment by KellieHawks** [comment]
Hey @lowip @chshapiro @hokstuff,
This guide Justin and I have built takes you on a step-by-step journey on setup best practices when first integrating the iOS SDK and its core components in your application. Hopefully, Justin and I will get this guide out before the end of the quarter. Please review when you have time and follow up with any questions! Thanks.

**Comment by lowip** [comment]
Hey @KellieHawks, is there a way to display the rendered HTML for `_docs/_hidden/private_betas/ios_sdk_integration.md`. I'm not sure where I need to navigate to see it.

**Comment by KellieHawks** [comment]
> Hey @KellieHawks, is there a way to display the HTML for `_docs/_hidden/private_betas/ios_sdk_integration.md`. I'm not sure where I need to navigate to see it.

Hey @lowip, if you add "/ios_sdk/" to the end of the deployment link you should be able to see the doc. Heres a direct link: https://braze-docs-kh-ios2-uiakyckxmld.herokuapp.com/docs/ios_sdk/

Give it ~5 minutes to load if it doesn't show the first time. Let me know if you have any questions!

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 1]
```diff
@@ -0,0 +1,722 @@
...
+```swift
+class BrazeManager: NSObject {
+  // 1
+  static let shared = BrazeManager()
+  
+  // 2
+  private let apikey = "YOUR-API-KEY"
+  
+  // 3
+  private var appboyOptions[String:Any]{
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:53]:
```suggestion
  private var appboyOptions: [String:Any] {
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 2]
```diff
@@ -0,0 +1,722 @@
...
+```objc
+@implementation BrazeManager
+ 
+// 1
++ (instancetype)shared {
+    static BrazeManager *shared = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        shared = [[BrazeManager alloc] init];
+        // Do any other initialisation stuff here
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:69]: Do we want to keep this comment? It doesn't seem that this part of code is mentioned anywhere after.

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 3]
```diff
@@ -0,0 +1,722 @@
...
+{% alert important %}
+Initialization should be done in the main thread.
+{% endalert %}
+
+{% subtabs global %}
+{% subtab Swift %}
+```swift
+func application(_ application: UIapplication, didFinishLaunchingWithOptions launchOptions:[UIApplication.LaunchOptionsKey:Any]?){
+  Appboy.start(withAPIKey: apikey, in: applciation, withLaunchOptions: launchOptions, withAppboyOptions: appboyOptions)
+}
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:110]:
```suggestion
func application(_ application: UIapplication, didFinishLaunchingWithOptions launchOptions:[UIApplication.LaunchOptionsKey:Any]?) {
  Appboy.start(withAPIKey: apikey, in: application, withLaunchOptions: launchOptions, withAppboyOptions: appboyOptions)
}
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 4]
```diff
@@ -0,0 +1,722 @@
...
+
+##### Handle Appboy Initialization in the AppDelegate.swift
+Next, navigate back to the `AppDelegate.swift` file and add the following code snippet in the AppDelegate's `didFinishLaunching...` method to handle the Appboy initialization from the `BrazeManager.swift` helper file. Remember, there is no need to add an `import AppboyUI` statement in the `AppDelegate.swift`.
+
+{% subtabs global %}
+{% subtab Swift %}
+
+```swift
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions:
+[UIApplication.LaunchOptionsKey: Any]?) -> Bool {
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:133]:
```suggestion
func application(
  _ application: UIApplication, 
  didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
) -> Bool {
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 5]
```diff
@@ -0,0 +1,722 @@
...
+
+  // 2 
+  UNUserNotificationCenter.current().requestAuthorization(option: options){ (granted, error) in
+  // 3 
+    Appboy.sharedInstance()?.pushAuthorization(fromUserNotificationCenter: granted)
+  }
+  
+  // 4 
+  UIApplications.shared.registerForRemoteNotificiations()
+}
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:204]:
```suggestion
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions:[UIApplication.LaunchOptionsKey:Any]?) {
  Appboy.start(withAPIKey: apikey, in: application, withLaunchOptions: launchOptions, withAppboyOptions: appboyOptions)
  // 1 
  let options: UNAuthorizationOptions = [.alert, . sound, .bagde]

  // 2 
  UNUserNotificationCenter.current().requestAuthorization(option: options) { (granted, error) in
  // 3 
    Appboy.sharedInstance()?.pushAuthorization(fromUserNotificationCenter: granted)
  }
  
  // 4 
  UIApplications.shared.registerForRemoteNotificiations()
}
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 6]
```diff
@@ -0,0 +1,722 @@
...
+// MARK - Push Notifications
+extension BrazeManager {
+  // 1 
+  func application(_ application: UIApplication,
+didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+
+  // 2 
+  Appboy.sharedInstance().?registerDeviceToken(deviceToken)
+  }
+}
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:263]:
```suggestion
extension BrazeManager {
  // 1 
  func application(
    _ application: UIApplication,
    didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
  ) {
    // 2 
    Appboy.sharedInstance().?registerDeviceToken(deviceToken)
  }
}
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 7]
```diff
@@ -0,0 +1,722 @@
...
+```
+{% endsubtab %}
+{% subtab Objective-C %}
+```objc
+// MARK - Push Notifications
+  // 1
+ - (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+  // 2
+  [[Appboy sharedInstance] registerDeviceToken:deviceToken];
+}
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:273]:
```suggestion
// MARK - Push Notifications
// 1
- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
  // 2
  [[Appboy sharedInstance] registerDeviceToken:deviceToken];
}
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 8]
```diff
@@ -0,0 +1,722 @@
...
+
+###### Step 3: Remote Notification Handling
+The Braze SDK can handle remote notifications that originate from Braze. Forward remote notifications to Braze; the SDK will automatically ignore push notifications that do not originate from Braze. Add the following method to your `BrazeManager.swift` file in the push notification extension.
+
+{% subtabs global %}
+{% subtab Swift %}
+```swift
+func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+  Appboy.sharedInstance()?.register(application, didReceiveRemoteNotification: userInfo, fetchCompletiionHandler: completionHandler)
+}
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:289]:
```suggestion
func application(
  _ application: UIApplication, 
  didReceiveRemoteNotification userInfo: [AnyHashable : Any], 
  fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void
) {
  Appboy.sharedInstance()?.register(
    application, 
    didReceiveRemoteNotification: userInfo, 
    fetchCompletiionHandler: completionHandler
  )
}
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 9]
```diff
@@ -0,0 +1,722 @@
...
+
+The Braze SDK can handle the response of push notifications that originate from Braze. Forward the response of the notifications to Braze; the SDK will automatically ignore responses from push notifications that do not originate from Braze. Add the following method to your `BrazeManager.swift` file:
+
+{% subtabs global %}
+{% subtab Swift %}
+
+```swift
+func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
+  Appboy.sharedInstance()?.userNotificationCenter(center, didReceive: response, withCompletionHandler: completionHandler)
+}
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:311]:
```suggestion
func userNotificationCenter(
  _ center: UNUserNotificationCenter, 
  didReceive response: UNNotificationResponse, 
  withCompletionHandler completionHandler: @escaping () -> Void
) {
  Appboy.sharedInstance()?.userNotificationCenter(
    center, 
    didReceive: response, 
    withCompletionHandler: completionHandler
  )
}
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 10]
```diff
@@ -0,0 +1,722 @@
...
+```
+{% endsubtab %}
+{% subtab Objective-C %}
+```objc
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
+  [[Appboy sharedInstance] userNotificationCenter:center didReceiveNotificationResponse:response withCompletionHandler:completionHandler];
+}
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:318]:
```suggestion
- (void)userNotificationCenter:(UNUserNotificationCenter *)center
didReceiveNotificationResponse:(UNNotificationResponse *)response 
         withCompletionHandler:(void (^)(void))completionHandler {
  [[Appboy sharedInstance] userNotificationCenter:center 
                   didReceiveNotificationResponse:response 
                            withCompletionHandler:completionHandler];
}
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 11]
```diff
@@ -0,0 +1,722 @@
...
+  // 2 
+  var userId: String?{
+    return user?.userID
+  }
+
+  // 3
+  func changeUser(_userId: String){
+    Appboy.sharedInstance()?.changeUser(userId)
+  }
+}
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:362]:
```suggestion
extension BrazeManager {
  // 1
  var user: ABKUser? {
    return Appboy.sharedInstance()?.user
  }

  // 2 
  var userId: String? {
    return user?.userID
  }

  // 3
  func changeUser(_ userId: String) {
    Appboy.sharedInstance()?.changeUser(userId)
  }
}
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 12]
```diff
@@ -0,0 +1,722 @@
...
+{% tab Step 1: Create Extension %}
+
+##### Create Extension for Analytics Code
+
+Next, you must enable your production code to log crucial analytics metrics to Braze without the need for excessive `import AppboyUI` statements. The production code should call into the `BrazeManager` to log all analytics. 
+
+Create an extension for your analytics code in your `Brazemanager.swift` file so it reads in a more organized manner as to what purpose is being served in the helper file, like so:
+
+```swift
+//MARK: -Analytics
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:404]:
```suggestion
//MARK: - Analytics
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 13]
```diff
@@ -0,0 +1,722 @@
...
+open func logCustomEvent(_ eventName: String, withProperties properties: [AnyHashable : Any]?)
+```
+
+__Matching Method__<br>
+Log custom events from the `Appboy` object to Braze. `Properties` is an optional parameter with a default value of nil. Custom events are not required to have properties but are required to have a name. 
+
+{% subtabs global %}
+{% subtab Swift %}
+```swift
+func logCustomEvent(_ eventName: String, withProperties properties: [AnyHashable: Any]? = nil){
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:430]:
```suggestion
func logCustomEvent(_ eventName: String, withProperties properties: [AnyHashable: Any]? = nil) {
```

File: _docs/_hidden/private_betas/ios_sdk_integration.md [hunk 14]
```diff
@@ -0,0 +1,722 @@
...
+extension Brazemanager {
+
+  // 1 
+  func displayContentCards(navigationController: UINavigationController?){
+      
+  // 2 
+      let contentCardsVc = ABKContentCardsTableViewController()
+      contentCardsVc.title = "Content Cards"
+      navigationController?.pushViewController(contentCardsVc, animated: true)
+  }
```
Comment (lowip) [inline _docs/_hidden/private_betas/ios_sdk_integration.md:686]:
```suggestion
  func displayContentCards(navigationController: UINavigationController?) {
      
    // 2 
    let contentCardsVc = ABKContentCardsTableViewController()
    contentCardsVc.title = "Content Cards"
    navigationController?.pushViewController(contentCardsVc, animated: true)
  }
```

**Review by lowip — DISMISSED** [review]
LGTM, I left some formatting / typos fix suggestions. Feel free to discard

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md [hunk 15]
```diff
@@ -0,0 +1,731 @@
...
+
+{% tabs local %}
+{% tab Create BrazeManager.swift %}
+
+##### Create BrazeManager.swift
+To build out your `BrazeManager.swift` file, create a new Swift file named _BrazeManager_ to add to your project at your desired location. Next, Replace `import Foundation` with `import AppboyUI` for SPM (`import Appboy_iOS_SDK` for cocoapods) and then create a `BrazeManager` class that will be used to host all Braze-related methods and variables. `Appboy_iOS_SDK`
+
+{% alert note %}
+- `BrazeManager` is an `NSObject` class and not a struct, so it can conform to ABK delegates such as the `ABKInAppMessageUIDelegate`.
+- The `Brazemanager` is a singleton class by design to ensure that only one instance of this class will be used. This is done to provide a unified point of access to the object.
```
Comment (malandr2) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md:37]:
```suggestion
- The `BrazeManager` is a singleton class by design to ensure that only one instance of this class will be used. This is done to provide a unified point of access to the object.
```

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md [hunk 16]
```diff
@@ -0,0 +1,731 @@
...
+{% endalert %}
+
+### Access User Variables and Methods
+
+{% tabs local %}
+{% tab Create User Variables and Methods %}
+
+##### Create User Variables and Methods
+
+Next, you will want easy access to the `ABKUser` variables and methods. Create an extension for your user code in the `Brazemanager.swift` file so it reads in a more organized manner as to what purpose is being served in the helper file, like so:
```
Comment (malandr2) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md:361]:
```suggestion
Next, you will want easy access to the `ABKUser` variables and methods. Create an extension for your user code in the `BrazeManager.swift` file so it reads in a more organized manner as to what purpose is being served in the helper file, like so:
```

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md [hunk 17]
```diff
@@ -0,0 +1,731 @@
...
+  return nil;
+}
+```
+{% endsubtab %}
+{% endsubtabs %}
+{% endtab %}
+{% endtabs %}
+
+{% alert checkpoint %}
+Proceed to compile your code and run your application. <br><br>Try sending yourself an in-app message. <br><br>In the `Brazemanager.swift` file, set a breakpoint at the entry of the example `ABKInAppMessageUIDelegate` method. Send yourself an in-app message and confirm the breakpoint is hit before advancing any further.
```
Comment (malandr2) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md:661]:
```suggestion
Proceed to compile your code and run your application. <br><br>Try sending yourself an in-app message. <br><br>In the `BrazeManager.swift` file, set a breakpoint at the entry of the example `ABKInAppMessageUIDelegate` method. Send yourself an in-app message and confirm the breakpoint is hit before advancing any further. 
```

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md [hunk 18]
```diff
@@ -0,0 +1,731 @@
...
+Add this snippet below the analytics section. Note that the `BrazeManager.swift` object is set as the delegate; this will be where the `BrazeManager.swift` file handles all the `ABKInAppMessageUIDelegate` methods. 
+
+{% alert important %}
+The `ABKInAppMessageUIDelegate` does not come with any required methods, but listed below is an example of one.
+{% endalert %}
+
+{% subtabs global %}
+{% subtab Swift %}
+```swift
+// Mark: - ABKInAppMessage UI Delegate
```
Comment (malandr2) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md:622]:
```suggestion
// MARK: - ABKInAppMessage UI Delegate
```

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md [hunk 19]
```diff
@@ -0,0 +1,731 @@
...
+
+1. An `ABKUser` object represents a known or anonymous user in your iOS application. Add a computed variable to retrieve the `ABKUser`; this variable will be reused to retrieve variables about the user.
+2. Query the user variable to easily access the `userId`. Among the other variables, the `ABKUser` object is responsible for (`firstName`, `lastName`, `phone`, `homeCity`, etc.)
+3. Set the user by calling `changeUser()` with a corresponding `userId`.
+
+{% subtabs global %}
+{% subtab Swift %}
+
+```swift
+// Mark: User
```
Comment (malandr2) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/ios_sdk_integration.md:371]:
```suggestion
// MARK: - User
```

**Review by lydia-xie — APPROVED** [review]
looks good to me!

**Pull request merged**